src/libguac/guacamole/client-fntypes.h

```diff
@@ -48,6 +50,16 @@
  */
 typedef int guac_client_free_handler(guac_client* client);
 
+/**
+ * Handler that will run before immediately before pending users are promoted
+ * to full users. The pending user socket should be used to communicate to the
+ * pending users.
+ *
+ * @param client
+ *     The client whose handler was invoked.
+ */
+typedef void guac_join_pending_handler(guac_client* client);
```

Comment:
Should this function have error-handling semantics, as well, in line with things like the join handler?

Reply:
Yeah, I guess I can just refuse to promote any users in a batch if the handler fails.

Reply:
Ok, I've added it. Currently there aren't any handlers that use it, but it's there.